### PR TITLE
[FLINK-27292][docs-zh] Translate the "Data Type Extraction" section of "Data Types" to Chinese

### PR DESCRIPTION
**Pull request by jjiey**
## What is the purpose of the change

[FLINK-27292](https://issues.apache.org/jira/browse/FLINK-27292)
Translate the "Data Type Extraction" section of "Data Types" to Chinese.

## Brief change log

Translate the "Data Type Extraction" section of "Data Types" to Chinese, the doc file is located in
'docs/content.zh/docs/dev/table/types.md'

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by jjiey** [comment]
Hi @ChengkaiYang2022 , could you help to review this? thank you.

**Comment by jjiey** [comment]
I finished the translation, and sorry that no one assigned this issue to me first.

I have asked for [it](https://issues.apache.org/jira/browse/FLINK-27292), can @wuchong or @MartijnVisser assign this to me? thank you.

**Comment by ChengkaiYang2022** [comment]
@jjiey could you a add picture of the part you translate?
Just like https://github.com/apache/flink/pull/19496#issuecomment-1115132533

**Comment by jjiey** [comment]
> @jjiey could you a add picture of the part you translate? Just like [#19496 (comment)](https://github.com/apache/flink/pull/19496#issuecomment-1115132533)

@ChengkaiYang2022 fine. here is the [link](http://localhost:1313/flink/flink-docs-master/zh/docs/dev/table/types/#%E6%95%B0%E6%8D%AE%E7%B1%BB%E5%9E%8B%E6%8F%90%E5%8F%96), and picture is as follows.

File: docs/content.zh/docs/dev/table/types.md [hunk 1]
```diff
@@ -1558,25 +1558,18 @@ COALESCE(TRY_CAST('non-number' AS INT), 0) --- 结果返回数字 0 的 INT 格
...
+数据类型提取
 --------------------
 
 {{< tabs "extraction" >}}
 {{< tab "Java/Scala" >}}
-At many locations in the API, Flink tries to automatically extract data type from class information using
-reflection to avoid repetitive manual schema work. However, extracting a data type reflectively is not always
-successful because logical information might be missing. Therefore, it might be necessary to add additional
-information close to a class or field declaration for supporting the extraction logic.
+在 API 中的很多地方，Flink 都尝试使用反射从类信息中自动提取数据类型，以避免重复手动定义 schema 的工作。但是，通过反射提取数据类型并不总是有效，因为可能会缺失逻辑信息。因此，可能需要在类或字段声明的附近添加额外信息以支持提取逻辑。
```
Comment (ChengkaiYang2022) [inline docs/content.zh/docs/dev/table/types.md:1566]: 在 API 中的很多地方，Flink 都尝试利用反射机制从类信息中自动提取数据类型，以避免重复地手动定义 schema 。但是，通过反射提取数据类型并不总是有效的，因为有可能会缺失逻辑信息。因此，可能需要在类或字段声明的附近添加额外信息以支持提取逻辑。

File: docs/content.zh/docs/dev/table/types.md [hunk 2]
```diff
@@ -1558,25 +1558,18 @@ COALESCE(TRY_CAST('non-number' AS INT), 0) --- 结果返回数字 0 的 INT 格
...
 
-The following table lists classes that can be implicitly mapped to a data type without requiring further information.
+下表列出了无需更多信息即可隐式映射到数据类型的类。
 
-If you intend to implement classes in Scala, *it is recommended to use boxed types* (e.g. `java.lang.Integer`)
-instead of Scala's primitives. Scala's primitives (e.g. `Int` or `Double`) are compiled to JVM primitives (e.g.
-`int`/`double`) and result in `NOT NULL` semantics as shown in the table below. Furthermore, Scala primitives that
-are used in generics (e.g. `java.util.Map[Int, Double]`) are erased during compilation and lead to class
-information similar to `java.util.Map[java.lang.Object, java.lang.Object]`.
+如果你打算在 Scala 中实现类，*建议使用包装类型*（例如 `java.lang.Integer`）而不是 Scala 的基本类型。如下表所示，Scala 的基本类型（例如 `Int` 或 `Double`）被编译为 JVM 基本类型（例如 `int`/`double`）并产生 `NOT NULL` 语义。此外，在泛型中使用的 Scala 基本类型（例如 `java.util.Map[Int, Double]`）在编译期间会被抹去，导致类信息类似于 `java.util.Map[java.lang.Object, java.lang.Object]`。
```
Comment (ChengkaiYang2022) [inline docs/content.zh/docs/dev/table/types.md:1570]: 如果你打算在 Scala 中实现类，*建议使用包装类型*（例如 `java.lang.Integer`）而不是 Scala 的基本类型。如下表所示，Scala 的基本类型（例如 `Int` 或 `Double`）会被编译为 JVM 基本类型（例如 `int`/`double`）并产生 `NOT NULL` 语义。此外，在泛型中使用的 Scala 基本类型时（例如 `java.util.Map[Int, Double]`），基本类型在编译期间会被抹去，导致类信息类似于 `java.util.Map[java.lang.Object, java.lang.Object]`。

**Review by ChengkaiYang2022 — COMMENTED** [review]
I left some comments about grammar problems. @fsk119  would you mind take a look?

**Comment by jjiey** [comment]
Thanks for @ChengkaiYang2022‘s review, I've finished the modification.

**Pull request merged**